src/components/OverflowLine/index.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState, useCallback } from 'react';
+import { shape, string, oneOfType, node } from 'prop-types';
+import Typography from '@material-ui/core/Typography';
+import Tooltip from '../Tooltip';
+
+/**
+ * A single line component within a row of the schemaTable.
+ * If the content text overflows, returns a line component with
+ * the text ellipsed and tooltip is used to display full text instead.
+ */
+function OverflowLine({ classes, content }) {
+  /**
+   * Track the textOverflow state depending on the component's
+   * width size relative to the text's length.
+   */
+  const [isTextOverflow, setIsTextOverflow] = useState(false);
+  /**
+   * Create a callback ref method to use to track the component's
+   * width measurements.
+   * (using a callback ref ensures that changes made to the ref
+   *  component can update the isTextOverflow state)
+   */
+  const measuredRef = useCallback(element => {
+    if (element !== null) {
+      if (element.scrollWidth > element.offsetWidth) {
+        setIsTextOverflow(true);
+      }
+    }
+  }, []);
+
+  if (isTextOverflow) {
+    return (
+      <Tooltip title={content}>
+        <div className={classes.line}>
+          <Typography
+            component="div"
+            variant="subtitle2"
+            ref={measuredRef}
+            noWrap>
+            {content}
+          </Typography>
+        </div>
+      </Tooltip>
+    );
+  }
+
+  return (
+    <Typography
+      component="div"
+      variant="subtitle2"
+      className={classes.line}
+      ref={measuredRef}>
+      {content}
+    </Typography>
+  );
+}

```

Comment:
STR:

1. Make the window size big enough so that the description fits fully in the row
2. Reload the page
3. Make the window small in width

Expected: The triple dots and tooltip would show
Actual: The triple dots and the tooltip does not show up

Reply:
Seems like fixing this issue will be a bit more complicated than I initially thought. We will need to use an external library or a custom hook (beyond the callback ref we are using here) in order to detect changes when the window size changes. The react documentation mentions that

> If you want to be notified any time a component resizes, you may want to use ResizeObserver or a third-party Hook built on it. 

Would it be okay to separate this into a new issue so that and work on it there while also not blocking other PRs that currently rely on this PR (ex. #69 & #70)

Reply:
An easier approach is to always have the tooltip but display the ellipsis when the window size is small. I think this would remove a lot of the complexity including the callback ref. Thoughts?

Reply:
That sounds good. Applied changes that now the description always displays a tooltip. And the text of title and description are ellipsed in cases of overflow.